wgsl/index.bs

```diff
@@ -9360,40 +9361,53 @@ The following algorithm describes how to compute these tags for a given function
 
 Note: The entire graph can be destroyed at this point. The tags listed above are all that we need to remember to analyze callers of this function.
 
-### Pointer Desugaring ### {#pointer-desguar}
+### Pointer Desugaring ### {#pointer-desugar}
 
 Each [=formal parameter|parameter=] of [=pointer type=] in the [=address
 spaces/function=] address space is desugared as a local variable declaration
 whose initial value is equivalent to dereferencing the parameter.
 That is, function address space pointers are viewed as aliases to a local
 variable declaration.
 
-Each [=let-declaration=] with an [=effective-value-type=] that is a [=pointer
+Each [=let-declaration=], *L*, with an [=effective-value-type=] that is a [=pointer
 type=] is desugared as follows:
-* The initializer expression of the declaration is recorded.
-* Each [=identifier=] that [=resolves=] is substituted with the recorded
-    initializer expression (wrapped in a
-    [[#parenthesized-expressions|parenthesized expression]]).
+* Visit each subexpression, *SE*, of the initializer expression of *L* in a postorder depth-first traversal:
```

Comment:
Right. It has to be post-order because you want to capture the most deeply nested item first.

Then I started thinking:  Does this have to specify left-to-right traversal as well?  I don't think it does.
Interestingly, the desugaring doesn't have to produce and equivalent *value*, just an equivalent uniformity property.
And the only mutable variable that can have its uniformity affected is a function-scope var.  And the analysis only updates that uniformity property in an assigment statement in the same function.  So this desugaring is sufficient.

If we actually wanted to capture the right value, we'd also have to capture the values and side effects of function calls.  That's a lot more work, and not needed here.